enhancements/api-review/custom-tags-aws.md

```diff
@@ -89,14 +90,20 @@ userTags that are specified in the infrastructure resource will merge with userT
 
 The userTags field is intended to be set at install time and is considered immutable. Components that respect this field must only ever add tags that they retrieve from this field to cloud resources, they must never remove tags from the existing underlying cloud resource even if the tags are removed from this field(despite it being immutable).
 
+In case of hosted control plane deployments, the userTags field is updated with latest updates requested by user. The merge logic gives higher precedence to userTags when there is duplicate tag found on the AWS resource.
+
 If the userTags field is changed post-install, there is no guarantee about how an in-cluster operator will respond to the change. Some operators may reconcile the change and change tags on the AWS resource. Some operators may ignore the change. However, if tags are removed from userTags, the tag will not be removed from the AWS resource.
 
+For the resources created and managed by hosted control plane, cluster api provider for aws reconciles the user tags on AWS resources. The hosted control plane updates the `infrastructure.config.openshift.io` resource to reflect new tags in `resourceTags`. The OpenShift operators, both core and non-core (managed by RedHat), reconcile the respective AWS resources created and managed by them. 
```

Comment:
Where do the CAPI controllers actually source from? The hosted control plane resource will copy them into the AWSCluster which is within the control plane, and not the guest cluster. So why do we need to adjust the infrastructure resource at all? Are there things beyond what the AWS CAPI provider manages that also support updating the tags?

Reply:
>Are there things beyond what the AWS CAPI provider manages that also support updating the tags?
The day2 operators would rely on tags information from infrastructure resource to add while creation or update existing resources in cloud provider context.

Reply:
Have you looked into how that would actually work in the context of CAPI? I don't think CAPI (an upstream project) will be looking at our infrastructure resource. So I'm interested in what component is going to look at the infrastructure resource, and then how does it pass that information to CAPI through it's upstream APIs?

And in this case, does the infrastructure resource actually need to be updated, or can HCP update CAPI directly? (since it's the one managing that already no?)

Reply:
Hypershift already reconciles the Infrastructure resource today. It copies `HostedCluster.spec.aws.resourceTags` to `status.PlatformStatus.AWS.ResourceTags`
https://github.com/openshift/hypershift/blob/52c3734fe1aacde8c70bfb1755ddf7b5f8e61735/support/globalconfig/infrastructure.go#L68-L79

However, for EC2 instances, tags specified on the NodePool + HostedCluster will be propagated to the AWSMachineTemplate which will cause nodes rollout as CAPI doesn't support day2 in-place updates of infra fields. This is why ROSA(ocm) doesn't allow day2 changes to nodePool tags.

there is an effort in upstream CAPI for in-place upgrades, see: https://github.com/kubernetes-sigs/cluster-api/issues/12291